test/functional/p2p_getaddr_caching.py

```diff
@@ -42,6 +44,13 @@ def addr_received(self):
 class AddrTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
+        # Start onion ports after p2p and rpc ports.
+        port = PORT_MIN + 2 * PORT_RANGE
```

Comment:
pretty sure the tests only pass by accident? Locally they fail for me:


```
./test/functional/test_runner.py p2p_getaddr_caching feature_proxy feature_bind_extra p2p_add_connections 

p2p_add_connections.py | ✓ Passed  | 7 s
feature_bind_extra.py  | ✖ Failed  | 1 s
feature_proxy.py       | ✖ Failed  | 1 s
p2p_getaddr_caching.py | ✖ Failed  | 20 s

```

The basic idea is that all port ranges for all tests are fully assigned once the test_runner starts. Using the same range for different tests is asking for collisions. 

One option to fix this would be to start a third range for "other" ports or just use one of the 12 ports assigned to each test (initially intended to be used by the test nodes itself).

Reply:
Oops sorry about that, will have a PR up shortly...